### DISCUSSION TITLE
Confusion around library usage

### DISCUSSION THREAD

**Discussion by PlugaruT** [cloudevents/sdk-python#217] [Q&A] [answered]
Hi, 

the usage of the lib might be a bit confusing at first, at least for me :). Especially when trying to work with non-http sinks, like Kafka. 

The Kafka `to_binary` converters seems to point to `abstract.CloudEvent` type, which is only defined inside `http.event` module. 
But then, how do I use Kafka as a sink for the events? Should I just use the `http.event.CloudEvent` class to build events and then convert them using the Kafka converters?

Shouldn't then the `http.event.CloudEvent` class be extracted into a `common` module or something?

**Comment by xSAVIKx** [comment]
Hey @PlugaruT. Thx for your feedback.

I totally agree that we have some packaging issues right now. Unfortunately that's some legacy we've got that's quite hard to resolve without completely refactoring everything (that'd be great actually, but just takes quite some time).

The additions to the library such as Kafka package were done while the core was there for a long time. I'd love to create a new version, but have very limited time resources right now

**Reply by PlugaruT** [reply]
Is there any JIRA or a place where I can have a look on the things that need to be done? I'm a bit packed at work these days, but I can try and squeeze some work here since we plan on using the lib extensively.

**Reply by xSAVIKx** [reply, marked as answer]
Thank you for your interest. Unfortunately the V2 roadmap wasn't even yet discussed. There are some open tasks in the issues here and I believe any further development is gonna be done in this platform
